6_drivers_prep_rmk.yml

```diff
@@ -0,0 +1,59 @@
+target_default: 6_drivers_prep
+
+packages:
+  - sp
+  - scipiper
+  - dplyr
+
+sources:
+  - 6_drivers_prep/code/nldas_cube_utils.R
+  - 6_drivers_prep/code/nldas_feather_utils.R
+  
+targets:
+  6_drivers_prep: 
+    depends: 
+      - 7_nldas_subset_rmk.yml
+      - 7_nldas_append_rmk.yml
+  
+  nldas_subset_time:
+    command: c(t0=I(0), t1 = I(145551))#345551
+    
+  nldas_time_chunk:
+    command: c(I(5000))
+    
+  nldas_cells:
+    command: list(x=I(c(221,221,344,344,223)), y=I(c(132,196,132,196,141)), variables=I(c('dlwrfsfc','pressfc','apcpsfc','vgrd10m','ugrd10m','dswrfsfc','spfh2m','tmp2m'))) # will come from code or spatial data...
+      
+  nldas_subset_box: # this is a FULL time pull
+    command: nldas_diff_box(nldas_cells, '6_drivers_prep/in/nldas_cells.rds') 
+      
+  nldas_append_box: # this is a PARTIAL time pull, where partial time could be 0 (complete)
+    command: nldas_update_box(nldas_cells, nldas_subset_box) 
+
+  nldas_append_time:
+    command: nldas_times_diff(nldas_subset_time, '6_drivers_prep/in/nldas_times.rds')
+    
+  nldas_subset_files:
+    command: calc_nldas_files(nldas_subset_box, nldas_subset_time, nldas_time_chunk)
+  
+  nldas_append_files:
+    command: calc_nldas_files(nldas_append_box, nldas_append_time, nldas_time_chunk)
+  
+  nldas_pull_cells:
+    command: nldas_diff_cells(nldas_cells, '6_drivers_prep/in/nldas_cells.rds') 
+    
+  nldas_cap_cells:
+    command: nldas_update_cells(nldas_cells, nldas_pull_cells, '6_drivers_prep/in/nldas_times.rds')
+  
+  nldas_subset_plan:
+    command: create_cube_task_plan(nldas_subset_files, nc_dir = I("7_drivers_fetch/out"), ind_dir=I('7_drivers_fetch/log'))
+
+  nldas_append_plan:
+    command: create_cube_task_plan(nldas_append_files, nc_dir = I("7_drivers_fetch/out"), ind_dir=I('7_drivers_fetch/log'))
+
+  "7_nldas_subset_rmk.yml":
+    command: create_cube_task_makefile(target_name, nldas_subset_plan, include = I("6_drivers_prep_rmk.yml"), packages = I(c('dplyr','ncdf4')), sources = '7_drivers_fetch/code/nccopy_utils.R')
+    
+  "7_nldas_append_rmk.yml":
```

Comment:
Will 7_nldas_subset_rmk.yml and 7_nldas_append_rmk.yml get looped through in 7_drivers_fetch.yml, or within  a recipe yet to be written within this 6_drivers_prep.yml file?

I'm thinking about this coupling between 6_ and 7_. It feels sorta messy to me, but I also see why you did it (there's a lot of prep work, so a full phase devoted to prep work doesn't seem inordinate, and the fetching depends on the prep but is also pretty file-rich).

One way to proceed from here is to add the recipes that call loop_tasks to the end of this 6_ file, and to add the combination recipes to the beginning of the 7_ file. In that case I lightly question whether 6_ is really a prep rather than a fetch step, because all the fetching is really being controlled in 6_. But I also see that breaking the files there, assuming 7_drivers_fetch.yml doesn't `include` 6_drivers_prep.yml`, gives you some new/different options for ways to write and test a combination phase that doesn't require a complete build of 7_nldas_subset_rmk.yml and 7_nldas_append_rmk.yml, though you can loop back to vizstorm-GIF for some ideas on how to avoid that requirement within a single .yml, as well.

Based on my discussion about WQP with Sam yesterday, I'm also wondering whether 6_ can avoid depending on 7_ for the iterative, additive data pulling approach I think we're taking with both WQP and NLDAS. Will the task planning here use the combined results from the 7_ fetch to determine which NLDAS cells need to be pulled in the next round? If so, then maybe 6_ will need to depend on 7_? If that's the case, I might advocate a little more strongly for combining those two phases into one.

Reply:
chewing on this...to be honest, I haven't spent much time thinking about the orchestration/linkage across the yamls, since my approach may change and I don't want to work on that part of the plumbing if it is in flux. What I am working on is the functions that build the file lists and get the data, then working on the next round of utilities that will use the raw .nc data. I think I will come back to this discussion when I have moved beyond those utilities and am happy with how they work across many files.